apps/bandwagon/models.py

```diff
@@ -45,10 +45,25 @@ def __set__(self, obj, value):
         cache.set(self.key(obj), value, two_days)
 
 
+class CollectionQuerySet(caching.CachingQuerySet):
+
+    def with_has_addon(self, addon_id):
+        """Annotate a collection with a `has_addon` property related to `addon_id`"""
+        has_addon = """
+            select 1 from addons_collections as ac
+                where ac.addon_id = %s and ac.collection_id = collections.id
+                limit 1"""
+
+        return self.extra(
+            select={'has_addon': has_addon},
+            select_params=(addon_id,))
+
+
 class CollectionManager(amo.models.ManagerBase):
 
     def get_query_set(self):
         qs = super(CollectionManager, self).get_query_set()
+        qs = qs._clone(klass=CollectionQuerySet)
```

Comment:
Not really but I figured that might be the safest way to not break the whole cache/transforming magic behind the scenes.

That actually turns the current query set into a `CollectionQuerySet` so that we can chain other filters with the new `with_has_addon` filter.

If someone has a better way of hooking that in, I'm all up for it but that's what I thought might be safest. I didn't really want to go all the way and just return `CollectionQuerySet`, I could try that actually but I'm not sure what that'd break.